main/src/main/java/org/mobilitydata/gtfsvalidator/validator/LocationHasStopTimesValidator.java

```diff
@@ -48,19 +46,41 @@ public class LocationHasStopTimesValidator extends FileValidator {
 
   private final GtfsStopTimeTableContainer stopTimeTable;
 
+  private final GtfsLocationGroupStopsTableContainer locationGroupStopTable;
+
   @Inject
   LocationHasStopTimesValidator(
-      GtfsStopTableContainer stopTable, GtfsStopTimeTableContainer stopTimeTable) {
+      GtfsStopTableContainer stopTable,
+      GtfsStopTimeTableContainer stopTimeTable,
+      GtfsLocationGroupStopsTableContainer locationGroupStopTable) {
     this.stopTable = stopTable;
     this.stopTimeTable = stopTimeTable;
+    this.locationGroupStopTable = locationGroupStopTable;
   }
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
+    Set<String> stopIdsInStopTimesandLocationGroupStops = new HashSet<>();
+    Set<String> locationGroupIdsInStopTimes = new HashSet<>();
+
+    for (GtfsStopTime stopTime : stopTimeTable.getEntities()) {
+      stopIdsInStopTimesandLocationGroupStops.add(stopTime.stopId());
+      locationGroupIdsInStopTimes.add(stopTime.locationGroupId());
+    }
+
+    for (String locationGroupId : locationGroupIdsInStopTimes) {
+      for (GtfsLocationGroupStops locationGroupStop : locationGroupStopTable.getEntities()) {
```

Comment:
I think you can use `locationGroupStopTable.byLocationGroupId(locationGroupId)` instead of this second loop.
Something like:
```
      var byLocationGroupId = locationGroupStopTable.byLocationGroupId(locationGroupId);
      for (var locationGroupStop: byLocationGroupId) {
        stopIdsInStopTimesandLocationGroupStops.add(locationGroupStop.stopId());
      }
```
This would insure that each GtfsLocationGroupStops entity is visited only once.

Reply:
More concise and likely to be faster, great find!